book/src/getting-started/1-qobjects-in-rust.md

```diff
@@ -0,0 +1,77 @@
+- C++ - The traditional back-end of Qt - A fast, low-level language with a strong type system. C++ offers a rich ecosystem, many Qt-specific libraries and bare-metal performance. The cost when using C++ is that it is slow to develop, very error-prone and can easily lead to memory-issues, which can instantly crash your application and cause security issues.
+
+Notably absent then is a back-end language that allows us to get rid of the issues C++ has and provides us with a safe way to write fast Back-end code.
+This of course is where Rust comes in.
+Whilst Rust doesn't have quite as rich of an ecosystem, it is typically faster to develop than C++, with easy dependency management, and most importantly, safe memory access.
+Therefore it is an ideal candidate to replace C++ for writing the back-end business-logic code that feeds the GUI with data.
+
+However, both C++ as well as QML still have their place in Qt applications.
+For that reason Rust, C++ and QML should all be able to be used to complement each other.
+CXX-Qt aims to make it easy to integrate all three languages easily with each other, through the use of Qts [meta object system](https://doc.qt.io/qt-5/metaobjects.html).
+
+Qts design is inherently object-oriented, which is true both for C++ and QML as well.
+Therefore, in order to integrate well with Qt, Rust needs to be able to extend the Qt object system with its own QObject subclasses and instances.
+This is exactly what CXX-Qt allows you to do.
+
+As Rust doesn't offer classes with inheritance and polymorphism, CXX-Qt uses a Rust module when defining new QObject subclasses.
+
+These CXX-Qt modules consist of multiple parts:
+- A `Data` struct
+    - Defines which Properties will be in the QObject subclass.
+    - Needs to implement the `Default` trait.
+    - This data will live in a C++ subclass that is generated from the CXX-Qt module
+- A `RustObj` struct
+    - A normal Rust struct
+    - One struct instance is created per class instance
+    - Contains any Rust-only data
+- The `impl` of the `RustObj` struct (optional):
+    - Contains any Rust code
+    - Functions marked with `#[invokable]` will be callable from QML and C++
+
+CXX-Qt will then expand this Rust module into two separate parts:
+1. A C++ subclass of QObject with the same name as the module
+2. The Rust struct `RustObj`
+
+<div style="padding: 1rem; text-align: center;">
+
+![Overview of CXX-Qt module generation](../images/cxx_qt_module_generation.png)
+
+</div>
+
+CXX-Qt also generates the code needed for interaction of the C++ QObject subclass and the `RustObj` struct using the [CXX library](https://cxx.rs/).
+For more details, see the [Concepts: Bridge](../concepts/bridge.md) page.
+CXX-Qt also wraps some Qt types for us, so they can be easily used by the Rust side.
+See the [Concepts: Qt Types](../concepts/types.md) page for the list of available types.
+
+The important take away here is the duality of any subclass generated by CXX-Qt.
```

Comment:
Do we need to note (or have you already?) that the properties defined on the Data struct actually end up as members on the C++ QObject and are owned there? Then RustObj needs to access the properties via the CppObj (this part can be mentioned later).

Reply:
Hm, then I didn't make that clear enough, will add a sentence about it in this paragraph.